### PR TITLE
Avoid swallowing exceptions, to make error reporting and debugging easier

### PR DESCRIPTION
**Pull request by quinton-hoole**
This PR looks biggish, but all it does is remove try catch blocks that just print out stack traces, and instead lets the exceptions propagate back to callers.  Removing the associated indentation makes the PR look bigger than it really is.

### CODE REVIEW

**Comment by quinton-hoole** [comment]
Oh, all checks and examples run correctly:

```
 ./gradlew examples:run

BUILD SUCCESSFUL in 4m 46s
35 actionable tasks: 35 executed
```

File: examples/minnietwitter/src/main/java/amino/run/appexamples/minnietwitter/MinnieTwitterMain.java [hunk 1]
```diff
@@ -71,75 +74,73 @@ private static void printStatistics() {}
      * (Sungwook Moon, 12/5/2017) This is a temporary method for demo to show shift policy. This
      * method adds a single user and send tweet messages for predefined times.
      */
-    private static void ExecuteSingleUserDemo(
-            InetSocketAddress hostAddr, InetSocketAddress omsAddr) {
+    private static void ExecuteSingleUserDemo(InetSocketAddress hostAddr, InetSocketAddress omsAddr)
```
Comment (VenuReddy2103) [inline examples/minnietwitter/src/main/java/amino/run/appexamples/minnietwitter/MinnieTwitterMain.java]: This method is not being used anymore. shall we remove it ?

File: examples/minnietwitter/src/main/java/amino/run/appexamples/minnietwitter/MinnieTwitterMain.java [hunk 2]
```diff
@@ -280,102 +281,99 @@ of detachFrom() just deletes the one and only reference to the microservice in O
...
-
-            System.out.println("Done populating.");
-        } catch (Exception e) {
-            e.printStackTrace();
         }
+
+        System.out.println("Done populating.");
     }
 
     public static void main(String[] args) {
```
Comment (VenuReddy2103) [inline examples/minnietwitter/src/main/java/amino/run/appexamples/minnietwitter/MinnieTwitterMain.java]: throw exception from this main too when runFullDemo throws exception ?

File: examples/minnietwitter/src/main/java/amino/run/appexamples/minnietwitter/MinnieTwitterMain.java [hunk 3]
```diff
@@ -280,102 +281,99 @@ of detachFrom() just deletes the one and only reference to the microservice in O
...
+        /*
+         * TODO quinton: The rest of this function is still really horrible code.
+         * It's the original UW code, that I've just moved here, but not had time to
+         * clean up properly yet.
+         */
+        /* Create the users */
+        List<User> users = new ArrayList<User>();
+        for (int i = 0; i < USER_COUNT; i++) {
+            long start = System.nanoTime();
+            User u = userManager.addUser("user" + i, "user" + i);
```
Comment (VenuReddy2103) [inline examples/minnietwitter/src/main/java/amino/run/appexamples/minnietwitter/MinnieTwitterMain.java]: Not related to this PR. we shall delete all the microservices created in runFullDemo() at the end of testcase. May be in separate PR.

File: examples/minnietwitter/src/main/java/amino/run/appexamples/minnietwitter/MinnieTwitterMain.java (hunk 1 above)
Comment (quinton-hoole) [reply examples/minnietwitter/src/main/java/amino/run/appexamples/minnietwitter/MinnieTwitterMain.java]: Yes, agreed.

File: examples/minnietwitter/src/main/java/amino/run/appexamples/minnietwitter/MinnieTwitterMain.java (hunk 3 above)
Comment (quinton-hoole) [reply examples/minnietwitter/src/main/java/amino/run/appexamples/minnietwitter/MinnieTwitterMain.java]: Agreed, will do in separate PR.

File: examples/minnietwitter/src/main/java/amino/run/appexamples/minnietwitter/MinnieTwitterMain.java (hunk 2 above)
Comment (quinton-hoole) [reply examples/minnietwitter/src/main/java/amino/run/appexamples/minnietwitter/MinnieTwitterMain.java]: Agreed.

**Comment by VenuReddy2103** [comment]
LGTM

**Comment by quinton-hoole** [comment]
Still busy wrapping this one up. Will update, rebase and retest this week.

File: examples/minnietwitter/src/main/java/amino/run/appexamples/minnietwitter/MinnieTwitterMain.java (hunk 1 above)
Comment (quinton-hoole) [reply examples/minnietwitter/src/main/java/amino/run/appexamples/minnietwitter/MinnieTwitterMain.java]: Done.

File: examples/minnietwitter/src/main/java/amino/run/appexamples/minnietwitter/MinnieTwitterMain.java (hunk 2 above)
Comment (quinton-hoole) [reply examples/minnietwitter/src/main/java/amino/run/appexamples/minnietwitter/MinnieTwitterMain.java]: Done.

**Comment by quinton-hoole** [comment]
Addressed comments, rebased, and retested.  All tests and examples pass. Merging.
